### PR TITLE
DBZ-1859 Enhance MongoDB connector metrics

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-1859

### CODE REVIEW

**Comment by Naros** [comment]
Per @jpechane, he suggests changing `MetadataEvent` to `ConnectorEvent` since we already use the term metadata elsewhere and it generalizes it a bit more and avoids overloading the naming.

File: debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/MongoMetricsIT.java [hunk 1]
```diff
@@ -128,6 +130,10 @@ public void testStreamingOnlyMetrics() throws Exception {
         final MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
         final ObjectName objectName = getStreamingMetricsObjectName("mongodb", "mongo1");
 
+        for (MBeanAttributeInfo info : mBeanServer.getMBeanInfo(objectName).getAttributes()) {
+            System.out.println(info.getName());
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/MongoMetricsIT.java:134]: Left-over?

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSourceMetrics.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import io.debezium.connector.base.ChangeEventQueueMetrics;
+import io.debezium.connector.common.CdcSourceTaskContext;
+import io.debezium.pipeline.ConnectorEvent;
+import io.debezium.pipeline.metrics.StreamingChangeEventSourceMetrics;
+import io.debezium.pipeline.source.spi.EventMetadataProvider;
+
+/**
+ * @author Chris Cranford
+ */
+public class MongoDbStreamingChangeEventSourceMetrics extends StreamingChangeEventSourceMetrics implements MongoDbStreamingChangeEventSourceMetricsMBean {
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSourceMetrics.java:19]: Might add the `@ThreadSafe` marker here and elsewhere potentially? I feel we got bit lenient about using those lately.

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/DisconnectEvent.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb;
```
Comment (gunnarmorling) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/DisconnectEvent.java:6]: Maybe move all the metrics stuff in this connector into its own sub-package?

**Comment by gunnarmorling** [comment]
Hey @Naros, do you think we can wrup this up for a release on Wednesday or Thursday? Thx!

**Comment by gunnarmorling** [comment]
Rebased and applied, thanks! Also pushed one more commit for adding the new metrics to the docs. Let's not forget about it :)

**Comment by Naros** [comment]
Ah; good point.

**Pull request merged**